### DISCUSSION TITLE
How to reduce bundle size?

### DISCUSSION THREAD

**Discussion by Parnswir** [gladly-team/next-firebase-auth#81] [Q&A] [answered]
Thanks to your awesome package, I could finally get all of my firebase-auth-related use-cases to work in Next. However, I noticed the bundle size (aka `next build`'s "First Load JS shared by all") has doubled from ~64k to ~133k just by following the tutorial.

Is this common/expected? Can you point to any obvious mistakes one can make?

**Comment by kmjennison** [comment, marked as answer]
This package's built code is 10kb on the client side and 21kb on the server side, so that difference probably includes other dependencies. Can you analyze the bundles before/after to see what's added?

**Reply by Parnswir** [reply]
hmm, you're right. It looks like firebase-auth's `auth.esm.js` is the culprit. So I guess there's no way around the larger bundle with your library or without. Thanks for the quick answer!

**Reply by Parnswir** [reply]
It looks like there is hope on the horizon: https://github.com/firebase/firebase-js-sdk/issues/4368
